### PR TITLE
Add code style check and build script

### PR DESCRIPTION
**Pull request by patrickrobrecht**
- Adds a code style check for CSS files using stylelint. Existing violations are resolved already.
- Adds a code style check for PHP files. Existing violations are fixed.
- Defines a build script for creating the `.min.css` files. The currently versioned `.min.css` files are removed.
- Changes the GitHub Actions to include a `composer install`. In addition, the branch is changed from `master` to `stable`. Before the next release we should rename `master` to `develop` and create a new `stable` branch.
- Add .distignore to deply minified CSS files and to ignore README.md (in facor of readme.txt)

### CODE REVIEW

File: .github/workflows/wordpress-plugin-asset-update.yml [hunk 1]
```diff
@@ -2,16 +2,17 @@ name: Plugin asset/readme update
 on:
   push:
     branches:
```
Comment (stklcode) [inline .github/workflows/wordpress-plugin-asset-update.yml:4]: We obviously habe to rename the branches before this works.

File: .github/workflows/wordpress-plugin-asset-update.yml [hunk 2]
```diff
@@ -2,16 +2,17 @@ name: Plugin asset/readme update
...
+    - stable
 jobs:
   master:
-    name: Push to master
+    name: Push to stable
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@master
+    - uses: actions/checkout@v2
+    - uses: php-actions/composer@v1
```
Comment (stklcode) [inline .github/workflows/wordpress-plugin-asset-update.yml]: Any reason for v1?

**Comment by stklcode** [comment]
We do use “GPLv3 or later“ consistently, also in the new _composer.json_, but the License URI points to GPL 2.0.

Domain path should also be redundant as we removed the language files.

And the GH branch no longer matches the deployment branch from the action.

https://github.com/pluginkollektiv/snitch/blob/07db7176aebbc7dbdc3e1c86e5b234cae8fb1b78/snitch.php#L8-L24

File: .github/workflows/wordpress-plugin-asset-update.yml (hunk 1 above)
Comment (patrickrobrecht) [reply .github/workflows/wordpress-plugin-asset-update.yml:4]: My plan is to create the `develop` branch directly after this PR is merged and `stable` when the next release is ready.

**Comment by patrickrobrecht** [comment]
> We do use “GPLv3 or later“ consistently, also in the new _composer.json_, but the License URI points to GPL 2.0.

Fixed.

> Domain path should also be redundant as we removed the language files.

Removed, should have been in the PR #40.
 
> And the GH branch no longer matches the deployment branch from the action.
> 
> https://github.com/pluginkollektiv/snitch/blob/07db7176aebbc7dbdc3e1c86e5b234cae8fb1b78/snitch.php#L8-L24

The two GitHub lines are removed completely.

**Review by stklcode — CHANGES_REQUESTED** [review]
Good work.

Think we do need a _.distignore_ for a fully functional workflow, though to get the minified resourced deployed and to not deploy the _README.md_

**Comment by patrickrobrecht** [comment]
> Think we do need a _.distignore_ for a fully functional workflow, though to get the minified resourced deployed and to not deploy the _README.md_

You are right, addressed with the latest commit.

**Review by stklcode — APPROVED** [review]
LGTM

**Pull request merged**